galley/pkg/config/processing/snapshotter/analyzingdistributor_test.go

```diff
@@ -241,6 +241,109 @@ func TestAnalyzeSortsMessages(t *testing.T) {
 	g.Expect(u.messages[1].Origin).To(Equal(o1))
 }
 
+func TestAnalyzeSuppressesMessages(t *testing.T) {
```

Comment:
Nit: these scenarios would really benefit strongly from table-oriented testing to avoid lots of duplicated scaffolding. 

I'm unwilling to block the PR on that, but I think it would make for clearer & nicer test code.

Reply:
Do you mean specifically the suppression testing, or all tests within the file?

Reply:
I was just thinking of the suppression testing, but looking closer it looks like it might make sense more broadly (at least for the various filtering functions). But that's definitely getting scope-creepy. :)

Reply:
I'm giving this a shot, just for the tests I've added. If a pattern emerges I'll be happy to apply it to the entire file.